contracts/test/StakeTable.t.sol

```diff
@@ -637,6 +656,596 @@ contract StakeTable_register_Test is LightClientCommonTest {
+        address validator2 = makeAddr("validator2");
+        registerValidatorOnStakeTable(validator1, seed1, COMMISSION, stakeTable);
+        registerValidatorOnStakeTable(validator2, seed2, COMMISSION, stakeTable);
+
+        vm.startPrank(delegator);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Delegated(delegator, validator1, INITIAL_BALANCE / 3);
+        stakeTable.delegate(validator1, INITIAL_BALANCE / 3);
+
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Delegated(delegator, validator2, INITIAL_BALANCE / 3);
+        stakeTable.delegate(validator2, INITIAL_BALANCE / 3);
+
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Undelegated(delegator, validator1, INITIAL_BALANCE / 3);
+        stakeTable.undelegate(validator1, INITIAL_BALANCE / 3);
+        (uint256 amountUndelegated, uint256 unlocksAt) =
+            stakeTable.undelegations(validator1, delegator);
+        assertEq(amountUndelegated, INITIAL_BALANCE / 3);
+        assertEq(unlocksAt, block.timestamp + ESCROW_PERIOD);
+        assertEq(token.balanceOf(address(stakeTable)), INITIAL_BALANCE / 3 + INITIAL_BALANCE / 3);
+        (uint256 validatorDelegatedAmount,) = stakeTable.validators(validator1);
+        assertEq(validatorDelegatedAmount, 0);
+
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Undelegated(delegator, validator2, INITIAL_BALANCE / 3);
+        stakeTable.undelegate(validator2, INITIAL_BALANCE / 3);
+        (amountUndelegated, unlocksAt) = stakeTable.undelegations(validator2, delegator);
+        assertEq(amountUndelegated, INITIAL_BALANCE / 3);
+        assertEq(unlocksAt, block.timestamp + ESCROW_PERIOD);
+        assertEq(token.balanceOf(address(stakeTable)), INITIAL_BALANCE / 3 + INITIAL_BALANCE / 3);
+        (validatorDelegatedAmount,) = stakeTable.validators(validator2);
+        assertEq(validatorDelegatedAmount, 0);
+
+        vm.warp(block.timestamp + ESCROW_PERIOD);
+        uint256 delegatorBalanceBefore = token.balanceOf(delegator);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Withdrawal(delegator, INITIAL_BALANCE / 3);
+        stakeTable.claimWithdrawal(validator1);
+        assertEq(token.balanceOf(delegator), delegatorBalanceBefore + INITIAL_BALANCE / 3);
+        assertEq(token.balanceOf(address(stakeTable)), INITIAL_BALANCE / 3);
+
+        delegatorBalanceBefore = token.balanceOf(delegator);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Withdrawal(delegator, INITIAL_BALANCE / 3);
+        stakeTable.claimWithdrawal(validator2);
+        assertEq(token.balanceOf(delegator), delegatorBalanceBefore + INITIAL_BALANCE / 3);
+        assertEq(token.balanceOf(address(stakeTable)), 0);
+
+        vm.stopPrank();
+    }
+
+    function test_RevertWhen_WithdrawFromNonExistentUndelegation() public {
```

Comment:
Was a bit confused by the test name

```suggestion
    function test_RevertWhen_UndelegateWithoutDelegation() public {
```

Reply:
Or should this test call `claimWithdrawal`?

Reply:
it should have tested claimWithdrawal https://github.com/EspressoSystems/espresso-network/commit/2d2e8e95c9c95c166f165d0e1330888a9abc1515